### ISSUE TITLE
Move count as an alternative to score calculation count

### ISSUE THREAD

**Issue by triceo** [TimefoldAI/timefold-solver#1038]
"Score calculation count" currently represents two concepts:

- How many scores were computed by the score director.
- How many moves were done by the solver.

Right now, both of these are the same thing, and the same number. The relationship of move to score calculation is 1:1. With the introduction of Ruin & Recreate moves, this changes; R&R move runs a nested CH, so for 1 R&R move, there are N score calculations. This needs to be reflected in solver reporting.

This is what needs to happen:

- [] Move count gets implemented similarly to score calculation count. (A Micrometer metric.) A move counts even if undone later. Undo moves do not count.
- [] Optionally, the metric also has sub-metrics per move type. (Some other metrics can already do that, for inspiration.)
- [] Solver logging additionally includes move speed (move count over time), along with score speed which it already does.
- [] Documentation uses move count where it previously used score calculation count.
- [] Benchmarker can generate a graph for this, just as it can for score calculation speed.

In short: move count/speed should be treated the same way as score count/speed. The same things should be possible, unless there are good reasons for them not to be.

Additional things to consider:

- [] Naming. Is "move count" and "move speed" the right name?
- [] Does this become the new metric by which we measure solver performance, and therefore is it enabled by default?If so, documentation needs to change, replacing uses of score calculation speed by move speed.

**Comment by zepfred** [comment]
> Naming. Is "move count" and "move speed" the right name?

For the benchmark report, it makes sense to me to present it as "Move Calculation Speed."

![Image](https://github.com/user-attachments/assets/400882f6-29f4-4a07-b2a1-18f5c85c2815)
![Image](https://github.com/user-attachments/assets/ab4a67d2-f308-43a5-b201-edec586526c8)

On the other hand, when defining the termination configuration, I think using only count is better: `moveCountLimit` and `unimprovedMoveCountLimit`.

I'll submit a pull request with this approach, and we can continue the discussion from there.
